server/api/routes/users.py

```diff
@@ -1,42 +1,86 @@
 from flask import Blueprint, request
 from flask_jwt_extended import create_access_token
 from database import User
+from api.routes.validators.users import valid_email, valid_password
 
 users = Blueprint('users', __name__)
 
-
 @users.route('/login', methods=['POST'])
 def login():
     """POST users/login
 
     Desc: Logs in a user
+          Ensures fields are not blank
+          Validates email used to log in
+          Checks if an account exists at all
+          Checks if an account exists and the password was incorrect
 
     Returns:
         200: success - true
-        401: success - false
+        401: password was incorrect, unauthorized access
+        404: no account was found
     """
     req = request.get_json()
-    user = User.objects(email=req['email'], password=req['password']).first()
-    if user:
-        access_token = create_access_token(identity=user)
+    if req['email'] == '' or req['password'] == '':
+        return {
+            'error': 'Fields must not be empty.'
```

Comment:
To make it easier on the frontend, I would recommend making these returns have a structure like this:
```
{
    'email': 'Email cannot be left blank',
    'password': 'Password cannot be left blank',
}
```
That way, the frontend team can simply receive the return data packet and place those specific error messages below each of the text input fields. Also, there can be different errors for each field. Not so much in the login, but in the registration it can be common.

Reply:
I meant this to apply to all the return packets with errors in both routes